File: php-laravel/lib/app/Models/RequestTimeFilterPostcodesArrivalSearch.php
<?php
/**
 * RequestTimeFilterPostcodesArrivalSearch
 */
namespace app\Models;

/**
 * RequestTimeFilterPostcodesArrivalSearch
 */
class RequestTimeFilterPostcodesArrivalSearch {

    /** @var string $id */
    private $id;

    /** @var \app\Models\RequestTransportation $transportation */
    private $transportation;

    /** @var int $travelTime */
    private $travelTime;

    /** @var \DateTime $arrivalTime */
    private $arrivalTime;

    /** @var \app\Models\RequestTimeFilterPostcodesProperty[] $properties */
    private $properties;

    /** @var \app\Models\RequestRangeFull $range */
    private $range;

}

File: scalaz/src/main/scala/org/openapitools/client/api/RequestTimeFilterPostcodesArrivalSearch.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import RequestTimeFilterPostcodesArrivalSearch._

case class RequestTimeFilterPostcodesArrivalSearch (
  id: String,
transportation: RequestTransportation,
travelTime: Integer,
arrivalTime: OffsetDateTime,
properties: List[RequestTimeFilterPostcodesProperty],
range: Option[RequestRangeFull])

object RequestTimeFilterPostcodesArrivalSearch {
  import DateTimeCodecs._

  implicit val RequestTimeFilterPostcodesArrivalSearchCodecJson: CodecJson[RequestTimeFilterPostcodesArrivalSearch] = CodecJson.derive[RequestTimeFilterPostcodesArrivalSearch]
  implicit val RequestTimeFilterPostcodesArrivalSearchDecoder: EntityDecoder[RequestTimeFilterPostcodesArrivalSearch] = jsonOf[RequestTimeFilterPostcodesArrivalSearch]
  implicit val RequestTimeFilterPostcodesArrivalSearchEncoder: EntityEncoder[RequestTimeFilterPostcodesArrivalSearch] = jsonEncoderOf[RequestTimeFilterPostcodesArrivalSearch]
}
